Source/WebKit/WebProcess/WebCoreSupport/WebPermissionController.cpp

```diff
@@ -72,18 +72,12 @@ void WebPermissionController::query(WebCore::ClientOrigin&& origin, WebCore::Per
 void WebPermissionController::tryProcessingRequests()
 {
     while (!m_requests.isEmpty()) {
-        auto& currentRequest = m_requests.first();
-        if (currentRequest.isWaitingForReply)
-            return;
-
-        currentRequest.isWaitingForReply = true;
-        WebProcess::singleton().sendWithAsyncReply(Messages::WebPermissionControllerProxy::Query(currentRequest.origin, currentRequest.descriptor, currentRequest.identifier, currentRequest.source), [this, weakThis = WeakPtr { *this }](auto state) {
-            RefPtr protectedThis { weakThis.get() };
-            if (!protectedThis)
-                return;
+        auto currentRequest = m_requests.takeFirst();
 
-            m_requests.takeFirst().completionHandler(state);
-            tryProcessingRequests();
```

Comment:
Will tryProcessingRequests() become dead code now? What did it do previously?

Reply:
In existing implementation, `tryProcessingRequests` just sends one async message for the first request in queue. If the first request has already sent message and no reply, it will return. That means one unhandled request will block the other requests in queue.
When the reply is received, the first request will be taken out of queue and tryProcessingRequests` is called to proceed next message.

We had this design because WebPermissionController used to cache permission result, which means `tryProcessingRequests` might handle multiple requests at a time (if the result is in the cache), and receiving one message reply (caching the result) may benefit the subsequent requests. 

But we removed the cache and send message for each request now, so we can send them when request is received, and finish CompletionHandler in message reply (as in this patch). And there is not need to store the requests in `m_requests`.